fedimint-core/src/bitcoin_migration.rs

```diff
@@ -0,0 +1,196 @@
+use std::str::FromStr;
+
+use bitcoin30::hashes::Hash as Bitcoin30Hash;
+use bitcoin_hashes::hex::FromHex;
+use bitcoin_hashes::Hash as Bitcoin29Hash;
+
+pub fn bitcoin29_to_bitcoin30_public_key(pk: bitcoin::PublicKey) -> bitcoin30::PublicKey {
+    bitcoin30::PublicKey::from_slice(&pk.to_bytes()).unwrap()
+}
+
+pub fn bitcoin29_to_bitcoin30_network(network: bitcoin::Network) -> bitcoin30::Network {
+    match network {
+        bitcoin::Network::Bitcoin => bitcoin30::Network::Bitcoin,
+        bitcoin::Network::Testnet => bitcoin30::Network::Testnet,
+        bitcoin::Network::Signet => bitcoin30::Network::Signet,
+        bitcoin::Network::Regtest => bitcoin30::Network::Regtest,
+    }
+}
+
+pub fn bitcoin30_to_bitcoin29_script(script: bitcoin30::ScriptBuf) -> bitcoin::Script {
+    bitcoin::Script::from_hex(&script.to_hex_string()).unwrap()
+}
+
+pub fn bitcoin30_to_bitcoin29_address(address: bitcoin30::Address) -> bitcoin::Address {
+    bitcoin::Address::from_str(&address.to_string()).unwrap()
```

Comment:
Same comment as above

Reply:
This seems a bit more interesting, need to check if the old version parsed taproot addresses correctly already.

Reply:
I checked and 0.29 already has taproot code, so I assume the both can always be converted between each other.

Reply:
Thanks for looking into that!